osu.Game/Collections/DrawableCollectionListItem.cs

```diff
@@ -117,11 +116,67 @@ private partial class ItemTextBox : OsuTextBox
         {
             protected override float LeftRightPadding => item_height / 2;
 
+            private const float count_text_size = 12;
+
+            [Resolved]
+            private RealmAccess realm { get; set; } = null!;
+
+            private readonly Live<BeatmapCollection> collection;
+
+            private OsuSpriteText countText = null!;
+
+            private IDisposable? itemCountSubscription;
+
+            public ItemTextBox(Live<BeatmapCollection> collection)
+            {
+                this.collection = collection;
+
+                CornerRadius = item_height / 2;
+            }
+
             [BackgroundDependencyLoader]
             private void load(OsuColour colours)
             {
                 BackgroundUnfocused = colours.GreySeaFoamDarker.Darken(0.5f);
                 BackgroundFocused = colours.GreySeaFoam;
+
+                if (collection.IsManaged)
+                {
+                    TextContainer.Height *= (Height - count_text_size) / Height;
+                    TextContainer.Margin = new MarginPadding { Bottom = count_text_size };
+
+                    TextContainer.Add(countText = new OsuSpriteText
+                    {
+                        Anchor = Anchor.BottomLeft,
+                        Origin = Anchor.TopLeft,
+                        Depth = float.MinValue,
+                        Font = OsuFont.Default.With(size: count_text_size, weight: FontWeight.SemiBold),
+                        Margin = new MarginPadding { Top = 2, Left = 2 },
+                        Colour = colours.Yellow
+                    });
+
+                    itemCountSubscription = realm.SubscribeToPropertyChanged(r => r.Find<BeatmapCollection>(collection.ID), c => c.BeatmapMD5Hashes, _ =>
```

Comment:
This is instant alarm bells for me. One subscription for each item on the list? So potentially N subscriptions for N items? There's no pooling or list virtualisation anywhere here, so that means one subscription for _every_ collection in db. It's also registered at `OsuGame` level, which means that the disposal of the subscription isn't going to save us any as it just _doesn't fire until the game is closed_.

I fear this is going to fall over _badly_ for power users with monstrous beatmap collections.

- Do we need this to be a subscription, even? The manage collections dialog is modal. There's not really any way for the counts to change while it's open except for background imports - but maybe it's fine not to support that?
- If we do need this to be a subscription, it should probably be handled a lot more economically. Maybe by eagerly acquiring on show and eagerly disposing on hide. Or by pooling/DLUW.

Reply:
It should be pretty efficient with https://github.com/realm/realm-dotnet/pull/3121, which is why I went for this.

I'd be more interested in moving it once we see it having an adverse effect. And definitely not by pooling (maybe one subscription at the dialog level, although that may have higher overhead due to no keypath filtering).

> Do we need this to be a subscription, even? The manage collections dialog is modal. There's not really any way for the counts to change while it's open except for background imports - but maybe it's fine not to support that?

It's modal, but unless the counts are refreshed each time it's shown, there's plenty of other ways the counts can change (right click menu at song select; collections dropdown at song select).

Reply:
Is the supposed performance concern still blocking this? I thought the flow was pushing stuff until someone complained.

Reply:
I'm not merging without testing. Someone linked a database with a bunch of huge collections so I might give this a try using that and see how hard it dies.

Reply:
Well I tested using database from https://github.com/ppy/osu/issues/27322#issuecomment-1960943814, which has ~80 collections. Opening the collections dialog increases the number of active subscriptions five-fold. I didn't see signs of falling over, surprisingly - even on mutations - but as predicted the subscriptions stay alive for the entirety of the remaining gameplay and the only thing that gets rid of them is deleting the collections. And 80 is probably not even that many collections.

I dunno, I'm still too skeptical to feel confident pushing this to users. It just feels way too shoddy to be keeping these subscriptions around when the dialog isn't even visible...